internal/deploy/helm/install.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package helm
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	"github.com/edgelesssys/constellation/v2/internal/logger"
+	"github.com/edgelesssys/constellation/v2/internal/retry"
+	"go.uber.org/zap"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/cli"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	// timeout is the maximum time given to the helm Installer.
+	timeout = 10 * time.Minute
+	// maximumRetryAttempts is the maximum number of attempts to retry a helm install.
+	maximumRetryAttempts = 3
+)
+
+// Installer is a wrapper for a helm install action.
+type Installer struct {
+	*action.Install
+	log *logger.Logger
+}
+
+// NewInstaller creates a new Installer with the given logger.
+func NewInstaller(log *logger.Logger, kubeconfig string) (*Installer, error) {
+	settings := cli.New()
+	settings.KubeConfig = kubeconfig
+
+	actionConfig := &action.Configuration{}
+	if err := actionConfig.Init(settings.RESTClientGetter(), constants.HelmNamespace,
+		"secret", log.Infof); err != nil {
+		return nil, err
+	}
+
+	action := action.NewInstall(actionConfig)
+	action.Namespace = constants.HelmNamespace
+	action.Timeout = timeout
+
+	return &Installer{
+		action,
+		log,
+	}, nil
+}
+
+// InstallChart is the generic install function for helm charts.
+// When timeout is nil, the default timeout is used.
+func (h *Installer) InstallChart(ctx context.Context, release Release) error {
+	return h.InstallChartWithValues(ctx, release, release.Values)
```

Comment:
```suggestion
	return h.InstallChartWithValues(ctx, release,  nil)
```
InstallChartWithValues already uses the release values, this makes the MergeMaps function more efficient.